build/bazelbuilder/Dockerfile

```diff
@@ -0,0 +1,63 @@
+FROM ubuntu:xenial-20170915
```

Comment:
I imagine we'll want to use a newer LTS version (and this is an opportune time). Asking around to see why we haven't upgraded the builder already and what version we're comfortable upgrading to if we do.

Reply:
Done. (Only updating the new image for now; if we want to update the old builder image, I can do that in a separate PR.)

Reply:
On second thought, this complicates things because `clang-3.9` isn't installable on newer LTS versions, and I was unsure that it made sense to have Bazel build with a different version of `clang` than everything else. I reverted that for now; we can always update them later.

Reply:
(Note that this will stop being a concern after we pin the C++ toolchains.)

Reply:
👍